### PR TITLE
Enhance app with integration test draft

### PR DESCRIPTION
**Pull request by qrutyy**
### This PR includes several important enhancements to the graph module, focusing on adding integration tests, refactoring, and new features to improve functionality and maintainability.

UI Testing Improvements & new features:

- Added tags to various components for better testability.
- Fixed the _FAQ_ box tag to differentiate between hoverable and non-hoverable states.
Code Refactoring:

- Moved graph-related classes to a dedicated package for improved structure and maintainability.

- Introduced a getter for edge direction type, making it easier to check and handle edge directions within the graph module.
Integration Tests:

- Implemented initial integration tests to verify critical functionality, ensuring that the system behaves as expected when different components interact.

- Updated the build script to include necessary dependencies for running integration tests.

**TODO:** 

- solve the issue with SQL import
- all TODO
- refactor
- linter
- fix a lil of ui (no.)
- make db creatable, when test execute

### CODE REVIEW

File: app/src/test/kotlin/integration/IntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Test
+    fun `graph is directed, check its view and FAQ button`() {
+        // I'm not sure if we can consider this as an integration test
+        // Imho it's an integration test, due to UI's connection with VM + there is more than 2 components (view/vm, or front/back)
+
+        // SETUP
+        val interactionSource = MutableInteractionSource()
+        val viewmodel = MainScreenViewModel(UndirectedGraph<Int>(),"UndirectedGraph")
+
```
Comment (spisladqo) [inline app/src/test/kotlin/integration/IntegrationTest.kt]: Why test says that graph is directed but then it creates an undirected graph?

Comment (qrutyy) [reply app/src/test/kotlin/integration/IntegrationTest.kt]: misprint

**Review by shakareem — APPROVED** [review]
Nice!

**Review by spisladqo — APPROVED** [review]
Good job!

**Pull request merged**